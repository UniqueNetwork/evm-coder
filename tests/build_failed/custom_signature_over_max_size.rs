#![allow(dead_code)]
use std::str::from_utf8;

use evm_coder::{
	make_signature,
	custom_signature::{SignatureUnit, SIGNATURE_SIZE_LIMIT},
};

trait Name {
	const SIGNATURE: SignatureUnit;

	fn name() -> &'static str {
		from_utf8(&Self::SIGNATURE.data[..Self::SIGNATURE.len]).expect("bad utf-8")
	}
}

impl<T: Name> Name for Vec<T> {
	evm_coder::make_signature!(new nameof(T) fixed("[]"));
}

struct MaxSize();
impl Name for MaxSize {
	const SIGNATURE: SignatureUnit = SignatureUnit {
		data: [b'!'; SIGNATURE_SIZE_LIMIT],
		len: SIGNATURE_SIZE_LIMIT,
	};
}

const NAME: SignatureUnit = <Vec<MaxSize>>::SIGNATURE;

fn main() {
	assert!(false);
}
