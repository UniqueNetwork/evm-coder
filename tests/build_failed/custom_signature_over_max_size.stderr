warning: unused import: `make_signature`
 --> tests/build_failed/custom_signature_over_max_size.rs:5:2
  |
5 |     make_signature,
  |     ^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error: any use of this value will cause an error
  --> tests/build_failed/custom_signature_over_max_size.rs:19:3
   |
18 |     const SIGNATURE: SignatureUnit =
   |     ------------------------------
19 |         evm_coder::make_signature!(new nameof(T::SIGNATURE) fixed("[]"));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ index out of bounds: the length is 256 but the index is 256
   |
   = note: `#[deny(const_err)]` on by default
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>
   = note: this error originates in the macro `$crate::make_signature` which comes from the expansion of the macro `evm_coder::make_signature` (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> tests/build_failed/custom_signature_over_max_size.rs:30:29
   |
30 | const NAME: SignatureUnit = <Vec<MaxSize>>::SIGNATURE;
   | -------------------------   ^^^^^^^^^^^^^^^^^^^^^^^^^ referenced constant has errors
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>
